sql-plugin/src/main/311until320-nondb/scala/com/nvidia/spark/rapids/shims/Spark31XShims.scala

```diff
@@ -71,6 +71,23 @@ abstract class Spark31XShims extends SparkShims with Spark31Xuntil33XShims with
     fileIndex.allFiles()
   }
 
+  override def alluxioReplacePathsPartitionDirectory(
+      pd: PartitionDirectory,
+      replaceFunc: Option[Path => Path]): (Seq[FileStatus], PartitionDirectory) = {
+    val updatedFileStatus = pd.files.map { f =>
+      val replaced = replaceFunc.get(f.getPath)
+      // Alluxio caches the entire file, so the size should be the same.
+      // Just hardcode block replication to 1 to make sure nothing weird happens but
```

Comment:
please clarify "nothing weird", why preserving the replication factor from the original FileStatus is a bad idea